pandas/tests/indexes/test_range.py

```diff
@@ -1037,3 +1037,27 @@ def test_engineless_lookup(self):
             assert idx.get_loc("a") == -1
 
         assert "_engine" in idx._cache
+
+    def test_reindex_limit(self):
+        # GH 28631
+        data = [["A", "A", "A"], ["B", "B", "B"], ["C", "C", "C"], ["D", "D", "D"]]
+        exp_data = [
+            ["A", "A", "A"],
+            ["B", "B", "B"],
+            ["C", "C", "C"],
+            ["D", "D", "D"],
+            ["D", "D", "D"],
+            [np.nan, np.nan, np.nan],
+        ]
+        df = pd.DataFrame(data)
+        result = df.reindex([0, 1, 2, 3, 4, 5], method="ffill", limit=1)
+        expected = pd.DataFrame(exp_data)
+        tm.assert_frame_equal(result, expected)
+
+    def test_get_indexer_limit(self):
+        # GH 28631
+        idx = Index(range(4))
+        target = Index([0, 1, 2, 3, 4, 5])
+        result = idx.get_indexer(target, method="pad", limit=1)
+        expected = np.array([0, 1, 2, 3, 3, -1])
+        assert (result == expected).all()
```

Comment:
could `tm.assert_numpy_array_equal` be used here?

Reply:
emm, i used this at first place, however, it failed on a certain version (py35 i recall) because of the type difference, one is `int64` and the other is `int32` if i remember correctly. So i take a trade-off to use current `assert`. If you have other suggestions on how to figure out type issue, I would be certainly happy to use `tm.assert_numpy_array_equal`!

Reply:
it failed on the windows tests? adding `dtype=np.intp` to expected should fix that.